lib/backends/webgl/ops/clip.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {Clip} from '../../../ops/clip';
+import {Tensor} from '../../../tensor';
+import {WebGLInferenceHandler} from '../inference-handler';
+import {ProgramInfo} from '../program-info';
+import {RunData} from '../program-manager';
+import {WebGLOperator} from '../webgl-operator';
+import {WebGLOperatorHelper} from '../webgl-operator-utils';
+
+export class WebGLClip extends Clip implements WebGLOperator {
+  run(inferenceHandler: WebGLInferenceHandler, inputs: Tensor[]): Tensor[] {
+    return WebGLOperatorHelper.run(this, inferenceHandler, inputs);
+  }
+  createProgramInfo(handler: WebGLInferenceHandler, inputs: Tensor[]): ProgramInfo {
+    const outputShape = inputs[0].dims.slice();
+    const shaderSource = `
+      const float min = float(${this.min});
+      const float max = float(${this.max});
+      uniform sampler2D A;
+      void main() {
+        float v = texture2D(A, TexCoords).r;
+        gl_FragColor = vec4(v < min ? min : (v > max) ? max : v);
```

Comment:
There is `clamp` function in WebGL specs at https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/clamp.xhtml . This function has the same semantic as Clip function here.

Reply:
Thanks a lot. Will take a look.